perfTest/recreateDb.js

```diff
@@ -0,0 +1,28 @@
+const { Pool } = require("pg");
+const { parse } = require("pg-connection-string");
+
+if (!process.env.PERF_DATABASE_URL) {
+  throw new Error(
+    "No PERF_DATABASE_URL setting detected; please don't call this script directly!"
+  );
+}
+const config = parse(process.env.PERF_DATABASE_URL);
+// don't connect to the provided db, or we can't drop it
+const pgPool = new Pool({ ...config, database: "postgres" });
+
+async function main() {
+  console.log(`Recreating database ${config.database}`);
+
+  const pgClient = await pgPool.connect();
+  try {
+    await pgPool.query(`drop database if exists ${config.database};`);
```

Comment:
```suggestion
    await pgClient.query(`drop database if exists ${pgClient.escapeIdentifier(config.database)};`);
```

Reply:
`pgClient` -> avoids a very unlikely but possible race condition due to pgPool potentially issuing two different clients which could cause synchronisation problems on a very overwhelmed database server.

`pgClient.escapeIdentifier` -> avoid SQL injection. Dropping a database based on envvars is already pretty bad, you might figure, but it's always worth limiting what someone can do to only what is expected.

Reply:
Done - I do know about sql injection 🙂, just wasn't sure how it would work given the value is parsed from a url. `escapeIdentifier` makes perfect sense though, thanks.